File: UiManagement/ConsoleUiManager.cpp
#include "./ConsoleUiManager.h"
#include <iostream>
#include <vector>
#include <string>
#include <cmath>

    void ConsoleUiManager::DisplayMessage(std::string message) const
    {
        std::cout << message;
    }
    void ConsoleUiManager::DisplayMessage(std::vector<std::string> &messages) const
    {
        for (std::string message : messages){
            std::cout << message << std::endl;
        }
    }

    std::string ConsoleUiManager::GetRawInput() const {
        std::string userInput;
        std::cin >> userInput;
        return userInput;
    }
    
    int ConsoleUiManager::GetIntInput() const {
        int userInput;
        std::cin >> userInput;
        return userInput;
        //changed it because it was easier that
        //way in the c++
    }

    ConsoleUiManager::ConsoleUiManager() {};

File: TavernSimCSharp/UiManagement/ConsoleUiManager.cs
public class ConsoleUiManager : UiManager
{
    public ConsoleUiManager () {}
    public void DisplayMessage(String message){
        Console.WriteLine(message);
    }
    public void DisplayMessage(String[] messages){
        foreach (String message in messages){
            DisplayMessage(message);
        }
    }
    public String GetRawInput(){
        return Convert.ToString(Console.ReadLine()); 
    }
    public int GetIntInput() {
        return Convert.ToInt32(Console.ReadLine());
    }
}